internal/e2ebenchmark/chain_client.go

```diff
@@ -0,0 +1,23 @@
+package e2ebenchmark
+
+import (
+	"testing"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+	"gotest.tools/v3/assert"
+
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	shardtypes "pkg.world.dev/world-engine/chain/x/shard/types"
+)
+
+type Chain struct {
+	shard shardtypes.QueryClient
+	bank  banktypes.QueryClient
+}
+
+func newChainClient(t *testing.T) Chain {
```

Comment:
Looks like chain client and nakama client in `e2ebenchmark` are the same as the clients in `e2e`, could we possibly combine the 4 clients into 2 clients and put them in a sibling package in `/internal`?

Reply:
I would say no to this. The code is copied over from e2e tests, but what will go in here and what these utlities will do is NOT e2e tests so even this could be modified. I repeat, this code is for benchmarking, and this client will very likely diverge from it's original e2e purpose. 

If we want to use some shared utility package I would say this is possible. But better to wait for a future PR where we know what we want more clearly. It's always better to make a design decision in the future when we actually have a practical issue to solve and understand rather then making a possibly wrong decision now and then possibly "refactoring" it again in the future.

Reply:
@namrapatel Same here, if you want me to just take this out. I can. But my rec is to leave it in.

Reply:
I think if the clients diverge in the future, then we should split them into 4 clients at that point. Currently, and seemingly in the near future, the 2 clients can be shared by `e2e` and `e2ebenchmark` both, so I'd be in favour of having a sibling package in `/internal` called something like `utils` or `clients` that can be shared by `e2e` and `e2ebenchmark`.

Would also be good to get @jerargus and/or @technicallyty to chime in on this one to get some more opinions

Reply:
maybe let's do this in another PR.

Reply:
Made ticket [here](https://linear.app/arguslabs/issue/WORLD-747/merge-client-code-in-e2e-and-e2ebenchmark), resolving

Reply:
@namrapatel created a ticket in the backlog, I will get to it when needed.